token/lib/_contractHelper.js

```diff
@@ -0,0 +1,212 @@
+    if (
+      !this.config.overrideOwnerWhitelist &&
+      !isValidOwner(networkId, newOwner, this.validOwners)
+    ) {
+      throw new Error(`${newOwner} is not a valid owner for the token contract`)
+    }
+    await this.ensureContractOwner(contract, sender)
+    const oldOwner = await contract.methods.owner().call()
+    if (oldOwner.toLowerCase() === newOwnerLower) {
+      throw new Error('old and new owner are the same')
+    }
+
+    const transaction = contract.methods.transferOwnership(newOwner)
+    await this.sendTransaction(networkId, transaction, { from: sender })
+    await withRetries(this.retries, async () => {
+      const ownerAfterTransaction =
+        (await contract.methods.owner().call()).toLowerCase()
+      if (
+        !this.config.multisig &&
+        ownerAfterTransaction !== newOwner.toLowerCase()
+      ) {
+        throw new Error(`New owner should be ${newOwner} but is ${ownerAfterTransaction}`)
+      }
+    }, this.config.verbose)
+  }
+
+  /**
+   * Sends an Ethereum transaction.
+   * @param {string} networkId - Ethereum network ID.
+   * @param {transaction} transaction - These are returned by contract.methods.MyMethod()
+   * @param {Object} opts - Options to be sent along with the transaction.
+   * @returns {Object} - Transaction receipt.
+   */
+  async sendTransaction(networkId, transaction, opts = {}) {
+    const web3 = this.web3(networkId)
+
+    if (!opts.from) {
+      opts.from = await this.defaultAccount(networkId)
+    }
+
+    if (this.config.multisig) {
+      // For multi-sig transactions, we submit the transaction to the multi-sig
+      // wallet instead of the token contract.
+      const contract = await this.contract(networkId)
+      transaction = await this.multiSigTransaction({
+        networkId,
+        sender: opts.from,
+        multiSigWalletAddress: this.config.multisig,
+        contractAddress: contract._address,
+        transaction
+      })
+    }
+
+    if (!opts.gas) {
+      opts.gas = await transaction.estimateGas({ from: opts.from })
+      this.vlog('estimated gas:', opts.gas)
+    }
+
+    // Send the transaction and grab the transaction hash when it's available.
+    this.vlog('sending transaction')
+    let transactionHash
+    transaction.send(opts)
+      .on('transactionHash', (hash) => {
+        transactionHash = hash
+        this.vlog('transaction hash:', transactionHash)
+      })
+
+    // Poll for the transaction receipt, with an exponential backoff. This works
+    // around some strange interactions between web3.js and some web3 providers.
+    // For example, this issue sometimes prevents transaction receipts from
+    // being returned when simply calling
+    // `await contract.methods.MyMethod(...).send(...)`:
+    //
+    // https://github.com/INFURA/infura/issues/95
+
+    // Blocks are mined every ~15 seconds, but it sometimes takes ~40-60 seconds
+    // to get a transaction receipt from rinkeby.infura.io.
+    const maxSleep = 120000
```

Comment:
What are maxSleep, totalSleep, sleepTime used for ? I did not see them referenced in the code.